src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -528,6 +529,15 @@ mongoc_client_pool_set_server_api (mongoc_client_pool_t *pool,
       return false;
    }
 
+   if (mongoc_client_pool_get_size (pool)) {
+      bson_set_error (error,
+                      MONGOC_ERROR_POOL,
+                      MONGOC_ERROR_POOL_API_TOO_LATE,
+                      "Cannot set server api after a client has been created");
+      return false;
+   }
+
    pool->api = mongoc_server_api_copy (api);
+   _mongoc_topology_scanner_set_server_api (pool->topology->scanner, api);
```

Comment:
Not sure if this is considered kosher or not. Please advise if this should be injected into the scanner through other means.